src/macros.rs

```diff
@@ -249,14 +249,22 @@ macro_rules! __log_module_path {
 #[macro_export]
 macro_rules! __log_file {
     () => {
-        file!()
+        if cfg!(feature = "track_caller") {
+            ::std::panic::Location::caller().file()
+        } else {
+            file!()
+        }
```

Comment:
This will break compilation of log with rustc < 1.46, even without the track_caller feature enabled. The following will allow compilation to succeed without that feature enabled:
```
        #[cfg(feature = "track_caller")]
        {
            ::std::panic::Location::caller().file()
        }
        #[cfg(not(feature = "track_caller"))]
        {
            file!()
        }
```